weather.py

```diff
@@ -1,72 +1,143 @@
+class Weather:
+    LocalObservationDateTime: str
+    EpochTime: int
+    WeatherText: str
+    WeatherIcon: int
+#    PrecipitationType: str
+    HasPrecipitation: bool
+    IsDayTime: bool
+    Temperature: Temperature
+    MobileLink: str
+    Link: str
+
+#data = {'LocalObservationDateTime': '2023-01-24T20:03:00+00:00', 'EpochTime': 1674590580, 'WeatherText': 'Mostly cloudy', 'WeatherIcon': 38, 'HasPrecipitation': False, 'PrecipitationType': None, 'IsDayTime': False, 'Temperature': {'Metric': {'Value': 6.7, 'Unit': 'C', 'UnitType': 17}, 'Imperial': {'Value': 44.0, 'Unit': 'F', 'UnitType': 18}}, 'MobileLink': 'http://www.accuweather.com/en/gb/leeds/ls1-6/current-weather/712327?lang=en-us', 'Link': 'http://www.accuweather.com/en/gb/leeds/ls1-6/current-weather/712327?lang=en-us'}
+
 def run():
-    # Define the api endpoint.
     api_endpoint = 'https://dataservice.accuweather.com'
 
-    # Define our api key. This is sensitive data, so we should consider passing
-    # it in as a flag, or reading from the environment.
-    apikey = 'bTWRzyxjP94K3Wn66NlQsr3h2qppp3cs'
-
-    # Define the api path for getting locations.
-    locations_path = 'locations/v1/regions'
-
-    # Here we call the get_locations() function passing in the locations url
-    # and api key.
-    locations = get_locations('{}/{}?apikey={}'.format(
-        api_endpoint, locations_path, apikey))
-
-    # ...and for example, let's say we want to know Africa's ID...
-    print(locations['Africa'])
-    # or Europe for that matter...
-    print(locations['Europe'])
-    # ...and let's pretend the user specifed the value as a command-line
-    # argument...
-    location = 'Central America'
-    print(locations[location])
-    # worth noting that what if the user specifed the location in lower-case?
-    # we could add all the locations to the dict in lower-case, then convert
-    # the user specifed location to lower-case too.
-    # Starting to sound like how to store stuff in a database. No coincidence!
+    apikey = '4bAzYrckrKgStAXpCiekZLyvkAGkHLmG'
+
+    # If user uses lower case,  we could add all the locations to the dict in lower-case, 
+    # then convert
+    
+    # Location to get the weather for
+    city = 'York'
+    # Admin area
+    admin_areas_path = 'locations/v1/adminareas'
+
+    admin_areas = get_locations('{}/{}?apikey={}'.format(
+        api_endpoint, admin_areas_path, apikey))
+
+    admin_area_code = (admin_areas[(city)])
+
+    # City
+    city_path = 'locations/v1/cities/search'
+    
+    search = '{} {}'.format(
+            city, admin_area_code)
+
+    url = '{}/{}?apikey={}&q={}'.format(
+        api_endpoint, city_path,  apikey, search )
+    
+    locations = make_request(url)
+
+    location: Location = dacite.from_dict(Location,locations)
+    location_key = location.Key
+
+    # Conditions
+    conditions_path = 'currentconditions/v1'
+
+    url = '{}/{}/{}?apikey={}'.format(
+        api_endpoint, conditions_path, location_key, apikey)
```

Comment:
Same here in regards to the `conditions_path`

Also, you keep reusing the `url` variable. I'd not do this as you are mutating the variable depending on the ordering of your code.

Instead create a new dict and add the urls to it:
```
urls = {
    'admin_areas' = '{}/{}'.format()   # whatever the formatting is for admin_areas
    'cities' = .....
}
```

Then when you make request you are explicitly passing in the url my name:

```
out = make_request(urls['admin_areas'])
```

Not only the the variable use cleaner, but you also benefit from knowing which url precisely you are passing to `make_request()`

Reply:
I've stopped mutating the `url` variable but I can't get the dictionary to work yet. The `conditions_url` depends on the `locations_url` having run already and retrieved the `location_key` Sure there will be a way around this. Also the `search` variable depends on `admin_areas` having run so I couldn't reference the first urls from a dictionary either.

Reply:
Ah, I think I see how the dictionary should look now but how do I pass in the variables to `format`?
`api_endpoint, apikey, search` for example?